### ISSUE TITLE
Ads credit post onboarding modals.

### ISSUE THREAD

**Issue by budzanowski** [woocommerce/pinterest-for-woocommerce#550]
After user has completed the setup and lands on the settings page he should be shown one of the following modals:

If billing setup is detected:
<img width="1484" alt="image" src="https://user-images.githubusercontent.com/17271089/191064459-be0d5a34-09e2-4645-a940-53e8e5a5d574.png">

If billing setup is not detected:
<img width="1484" alt="image" src="https://user-images.githubusercontent.com/17271089/191064558-56f82779-531f-4ae0-8cd4-6ac48fe470f7.png">

This is depends on https://github.com/woocommerce/pinterest-for-woocommerce/pull/549

## Technical Details

Add billing link should point to (in a new window tab)

[https://ads.pinterest.com/advertiser/{advertiser_id}/billing/](https://ads.pinterest.com/advertiser/%7Bavertiser_id%7D/billing/)

    If the original window/tab "gets focus" again check if billing is setup and poll every X seconds (open to discussion) to check again
    Once billing is setup, check eligibility and redeem
    Automatically close modal and display success message on catalog page

Google listings and Ads has prior art for modals/guides like this including handling for tracks events.

https://github.com/woocommerce/google-listings-and-ads/blob/develop/js/src/dashboard/campaign-creation-success-guide/index.js
https://github.com/woocommerce/google-listings-and-ads/tree/develop/js/src/components/app-modal
https://github.com/woocommerce/google-listings-and-ads/tree/develop/js/src/components/app-button

## Event Tracking

TBC - Use placeholders for event names/context for initial development

* We will want to track events when modal is opened/closed(dismissed)

**Comment by findingsimple** [comment]
Updated the link that can be used for billing and added some links to prior art within GLA for modals/guides and app buttons

**Comment by budzanowski** [comment]
@alexiglesias31 I have merged the banner svg graphics into `feature/ads-credit-flow` branch under:
`pinterest-for-woocommerce/assets/images/gift_banner.svg`;

**Comment by alexiglesias31** [comment]
@jconroy @budzanowski
A question related to this task. The open modal should be opened only the first time that user lands on the Catalog page? Or every X seconds? 

> If the original window/tab "gets focus" again check if billing is setup and poll every X seconds (open to discussion) to check again

This got my confused.

If you prefer we can handle this in a separate PR tho, and we can let this PR handling the modal component and the tracking events.

**Comment by budzanowski** [comment]
@alexiglesias31 I will let @jconroy answer that - I also don't fully understand 😕 

Lets just add modals here and they should open each time user lands in catalog. This should unblock you. The exact rules we will do in a separate Issue/PR.

**Comment by alexiglesias31** [comment]
Thank you @budzanowski, that's exactly what I was asking :blush:

**Comment by budzanowski** [comment]
Remaining work will be tracked in #559

**Comment by findingsimple** [comment]
> The open modal should be opened only the first time that user lands on the Catalog page? Or every X seconds?
> If the original window/tab "gets focus" again check if billing is setup and poll every X seconds (open to discussion) to check again
> This got my confused.

The way I presented in my [original issue](https://github.com/woocommerce/pinterest-for-woocommerce/issues/516) before it was split possibly makes a little more sense then above 

* If billing setup is not detected display relevant modal
* Link the merchant to [https://ads.pinterest.com/advertiser/{advertiser_id}/billing/](https://ads.pinterest.com/advertiser/%7Bavertiser_id%7D/billing/) in a new window tab
* If the original window/tab "gets focus" again check if billing is setup and poll every X seconds (open to discussion) to check again
* Once billing is setup, check eligibility and redeem

However we'll now simplify things and just close the modal and avoid the need to do any polling. Instead we'll just pick up changes with the billing background check.

I've added more info to the new issue here to try and clarify further (also designs have been tweak to add handling for errors/edge cases) https://github.com/woocommerce/pinterest-for-woocommerce/issues/559#issuecomment-1259014920
